### DISCUSSION TITLE
Child to Parent Conditional State Transitions

### DISCUSSION THREAD

**Discussion by corsali-albert** [statelyai/xstate#2315] [Q&A] [answered]
What are best practices for transitioning conditionally from a child state to a parent state?

In particular (codesandbox here if you don't want to read a text description https://codesandbox.io/s/bold-vaughan-wk7zh?file=/src/index.tsx), I have `machineA` and its two states `I` and `J`. `J` has submachine `machineB` with states `L`, `M`. When in state `J.M`, on `eventX`, I want to transition to `I` if some context-dependent condition is met `guard1`, otherwise I want to transition to `L`.

I was thinking I could define a transition on `machineA`, but then I don't think I could specify "only `J.M`" (i.e. it would also happen if the guard condition were met for `J.L`), correct?

**Comment by davidkpiano** [comment, marked as answer]
As mentioned in the Discord, you want to target the parent (or any other arbitrary) state node by its ID: https://xstate.js.org/docs/guides/ids.html#identifying-state-nodes

**Reply by corsali-albert** [reply]
Thanks! I think my more fundamental issue was just a suboptimal model design -- I refactored it and didn't even need that type of transition any more!

**Reply by davidkpiano** [reply]
Even better!
